src/PhpSpreadsheet/Cell/DataValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace PhpOffice\PhpSpreadsheet\Cell;
+
+use PhpOffice\PhpSpreadsheet\Calculation;
+use PhpOffice\PhpSpreadsheet\Calculation\Functions;
+
+class DataValidator
+{
+    /**
+     * Does this cell contain valid value?
+     *
+     * @param Cell $cell Cell to check the value
+     *
+     * @return bool
+     */
+    public function isValidCellValue(&$cell)
+    {
+        $cellValue = $cell->getValue();
+        $dataValidation = $cell->getDataValidation();
+
+        if (!$dataValidation->getAllowBlank() && ($cellValue === null || $cellValue == '')) {
+            return false;
+        }
+
+        // TODO: write check on all cases
+        switch ($dataValidation->getType()) {
+            case DataValidation::TYPE_LIST:
```

Comment:
Let's refactor the code for list validation in a private method. Something like "isValueInList()" or similar...